src/integrations/room-presence/room-presence.proxy.ts

```diff
@@ -3,10 +3,14 @@ import {
   STATE_NOT_HOME,
 } from './room-presence-distance.sensor';
 import { DeviceTracker } from '../../entities/device-tracker';
+import { Sensor } from '../../entities/sensor';
 
-export class RoomPresenceDeviceTrackerProxyHandler
+export class RoomPresenceProxyHandler
   implements ProxyHandler<RoomPresenceDistanceSensor> {
-  constructor(private readonly deviceTracker: DeviceTracker) {}
+  constructor(
+    private readonly deviceTracker: DeviceTracker,
+    private readonly batterySensor: Sensor = undefined
```

Comment:
Instead of initializing with `undefined` I would use the optional syntax:

```suggestion
    private readonly batterySensor?: Sensor
```

Reply:
Done